src/system/game-data.ts

```diff
@@ -2103,13 +2103,17 @@ export class GameData {
     };
   }
 
-  getStarterSpeciesDefaultAbilityIndex(species: PokemonSpecies): number {
-    const abilityAttr = this.starterData[species.speciesId].abilityAttr;
+  getStarterSpeciesDefaultAbilityIndex(species: PokemonSpecies, abilityAttr?: number): number {
+    if (isNullOrUndefined(abilityAttr)) {
+      abilityAttr = this.starterData[species.speciesId].abilityAttr;
+    }
```

Comment:
```suggestion
    abilityAttr ??= this.starterData[species.speciesId].abilityAttr;
```
You should be able to simplify it like this. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment